presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestingSqlServerServer.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.sqlserver;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.testing.TemporaryDockerContainer;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+import java.time.Duration;
+
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * This suite is required to run any Oracle Test as it provides docker container with Oracle.
+ */
+public final class TestingSqlServerServer
+{
+    private static final Logger LOG = Logger.get(TestingSqlServerServer.class);
+
+    public static final String JDBC_URL = "jdbc:sqlserver://localhost";
+    public static final String USER = "sa";
+    public static final String PASSWORD = "SQLServerPass1";
+
+    private static TemporaryDockerContainer dockerContainer;
+
+    @BeforeSuite
+    public void setupTestingOracleServer()
+            throws Exception
+    {
+        this.dockerContainer = new TemporaryDockerContainer(
+                "microsoft/mssql-server-linux:2017-CU6",
+                "sqlserver-master",
+                ImmutableList.of(1433),
+                ImmutableMap.of(
+                        "ACCEPT_EULA", "Y",
+                        "SA_PASSWORD", "SQLServerPass1"));
+
+        RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
+                .withMaxDuration(Duration.of(5, MINUTES))
+                .withMaxAttempts(Integer.MAX_VALUE) // limited by MaxDuration
+                .onRetry(event -> LOG.info("Waiting for sql-server..."))
+                .withDelay(Duration.of(5, SECONDS));
+        Failsafe.with(retryPolicy).run(() -> executeInSqlServer("SELECT 1"));
+    }
+
+    public static void executeInSqlServer(String sql)
+    {
+        try (Connection conn = DriverManager.getConnection(JDBC_URL, USER, PASSWORD);
+                Statement stmt = conn.createStatement()) {
+            stmt.execute(sql);
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Failed to execute statement: " + sql, e);
+        }
+    }
+
+    @AfterSuite(alwaysRun = true)
+    public void close()
+    {
+        dockerContainer.close();
+        dockerContainer = null;
+    }
+
+    @Test
+    public void putBreakPointHereAndThenRunOracleTestsj()
+    {
+        LOG.info("SqlServer docker container is up and running");
```

Comment:
This means the unit tests are not unit. And they are not product tests either.
This is Yet Another Way of Running Tests.

I feel strongly about this.

We should find our better version of Dockerized RDBMS which allows to use instance per test class. And make SQL Server unit tests _real unit tests_.